### PR TITLE
Document possible value type

### PR DESCRIPTION
**Pull request by greg0ire**
It seems that providing a single field name is ok.

### CODE REVIEW

File: lib/Elastica/Aggregation/TopHits.php [hunk 1]
```diff
@@ -66,7 +66,7 @@ public function setSort(array $sortArgs)
     /**
      * Allows to control how the _source field is returned with every hit.
      *
-     * @param array|bool $params Fields to be returned or false to disable source
+     * @param array|string|bool $params Fields to be returned or false to disable source
```
Comment (ruflin) [inline lib/Elastica/Aggregation/TopHits.php]: Interesting. I assume you tested that this works? Could we add a test case?

Comment (greg0ire) [reply lib/Elastica/Aggregation/TopHits.php]: I'll tried to, and yeah, I built my app that way but phpstan told me I couldn't

**Comment by ruflin** [comment]
Any chance you could add a test like here https://github.com/ruflin/Elastica/blob/master/test/Elastica/Aggregation/TopHitsTest.php#L288 to the PR that tests setSource without an array?

**Comment by greg0ire** [comment]
@ruflin sure, I was a bit overwhelmed lately but now I have the time

**Comment by greg0ire** [comment]
Didn't manage to test locally, but Travis seems to be happy with this

File: test/Elastica/Aggregation/TopHitsTest.php [hunk 2]
```diff
@@ -279,13 +279,22 @@ public function testAggregateTwoDocumentPerTagWithOffset()
         $this->assertEquals([2, 4], $resultDocs);
     }
 
+    public function limitedSourceProvider()
+    {
+        return [
+            'string source' => ['title'],
```
[the remark this answers is not in the document]
Comment (greg0ire) [reply test/Elastica/Aggregation/TopHitsTest.php]: No it should not be: this is not the first argument, this is an array of all the arguments, and the first one is a string, not an array.

File: test/Elastica/Aggregation/TopHitsTest.php [hunk 3]
```diff
@@ -279,13 +279,22 @@ public function testAggregateTwoDocumentPerTagWithOffset()
...
+            'array source' => [['title']],
```
Comment (greg0ire) [inline test/Elastica/Aggregation/TopHitsTest.php]: @ruflin here is how you pass an array, see the difference?

File: test/Elastica/Aggregation/TopHitsTest.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (ruflin) [reply test/Elastica/Aggregation/TopHitsTest.php]: Got it, reading https://phpunit.de/manual/current/en/writing-tests-for-phpunit.html#writing-tests-for-phpunit.data-providers also helped :-)

**Pull request merged**